### ISSUE TITLE
doubt about the generation of embeddings.

### ISSUE THREAD

**Issue by ErikJhones** [Graph-COM/PEG#2]
Hello. I liked your article. I would like to ask a question about the notebook test function inside the PEG_layer directory. In the test() function, you pass the test set to generate the structure that will be used to generate the embeddings, however, this seems a bit strange to me. Generally, the methods exclusively use the pos_train_edge_index set in the convolution step, leaving the test and validation sets only to generate the embeddings of the edges based on the structure generated by the pos_train_edge_index. Is there any reason for you to do it differently? If the reason is in the paper, then forgive me for not finding it.

**Comment by zoom-wang112358** [comment]
Hi. I think your question is about the test function in link_prediction_example.ipynb inside the PEG_layer directory. It's an example to show how to use PEG_layer. In that example, we use all the train_pos_edge_index to generate positional encodings. I'm not sure whether I understand your question. We also use the pos_train_edge_index set in the convolution step. We test our model on pos_val_edge_index and pos_test_edge_index (with negative sampling).

**Comment by ErikJhones** [comment]
Hello. Thanks for answer. My question is about the forward in the Net() class in the link_prediction_example.ipynb. 

    def forward(self, x, pos, pos_edge_index, neg_edge_index):
        edge_index = torch.cat([pos_edge_index, neg_edge_index], dim=-1)
        x = self.conv1(x, pos, pos_edge_index)
        x = self.conv2(x, pos, pos_edge_index)
        
here, self.conv1 and 2 must receive only the pos_train_edge_index. Passing the pos_val or pos_test doesn't seem right to me. But, in the test(), you pass pos_val_edge_index or pos_test_edge_index.

@torch.no_grad()
def test():
    model.eval()
    perfs = []
    for prefix in ["val", "test"]:
        pos_edge_index = data[f'{prefix}_pos_edge_index']
        neg_edge_index = data[f'{prefix}_neg_edge_index']

        link_logits = model(x, pos, pos_edge_index, neg_edge_index) # *****HERE *****

**Comment by ErikJhones** [comment]
In the test, it is usually passed pos_train_edge_index to self.conv, and the resulting embeddings are used to form the logits of the training and validation edges

**Comment by zoom-wang112358** [comment]
Hi. Thank you for pointing out. I made a mistake when writing this example. You are correct. We should always pass pos_train_edge_index to self.conv in the link prediction task generally, and then test the model over the val and test edge index (both positive and negative). I have updated the link_prediction_example.ipynb. Sorry for the confusion.
